amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -574,26 +590,95 @@ def generate_json_and_cbor_obj_for_write(file, format_option, binary):
 
 
 def clean_up():
-    if os.path.exists(output_file):
-        os.remove(output_file)
+    if os.path.exists(output_file_for_benchmarking):
+        os.remove(output_file_for_benchmarking)
+
+
+def output_result_table(results_output, table):
+    if results_output is None:
+        print(tabulate(table, tablefmt='fancy_grid'))
+    else:
+        result_list = []
+        fields = table[0]
+        for row in table[1:]:
+            obj = {field: value for field, value in zip(fields, row)}
+            result_list.append(obj)
+        # Creates the destination directory
+        des_dir = os.path.dirname(results_output)
+        if des_dir != '' and des_dir is not None and not os.path.exists(des_dir):
+            os.makedirs(des_dir)
+        # Creates the destination file
+        with open(results_output, 'bw') as fp:
+            ion.dump(result_list, fp, binary=False)
+        return result_list
+
+
+def compare_benchmark_results(previous_path, current_path, output_file_for_comparison):
+    with open(previous_path, 'br') as p, open(current_path, 'br') as c:
+        previous_results = ion.load(p)
+        current_results = ion.load(c)
+        # Creates a list to hold compared scores
+        results = []
+        # For results of each configuration pattern with the same file
+        for idx, prev_result in enumerate(previous_results):
+            cur_result = current_results[idx]
+            file_name = cur_result['file_name']
+            result = {'input': file_name,
+                      'command': cur_result['command'],
+                      'options': cur_result['options']}
+            relative_difference_score = {}
+            for keyword in BENCHMARK_SCORE_KEYWORDS:
+                cur = float(cur_result[keyword])
+                prev = float(prev_result[keyword])
+                relative_difference_score[keyword] = (cur - prev) / prev
+            result['relative_difference_score'] = relative_difference_score
+            results.append(result)
+        with open(output_file_for_comparison, 'bw') as o:
+            ion.dump(results, o, binary=False)
+
+        # check if regression happens
+        regression_file = has_regression(results)
+        if regression_file:
+            print(regression_file)
+        else:
+            print('no regression detected')
+        return regression_file
 
 
-def options_validation(format_options):
-    return True
+def has_regression(results):
+    for each_result in results:
+        relative_difference_score = each_result['relative_difference_score']
+        for field in relative_difference_score:
+            value_diff = relative_difference_score[field]
+            # TODO simply set the threshold to 1. Need optimization.
+            if value_diff > REGRESSION_THRESHOLD:
```

Comment:
Is 0 a safer value to choose in the meantime? To exceed 1, wouldn't the new version have to be twice as slow?